api/src/main/java/jakarta/mail/Multipart.java

```diff
@@ -266,4 +259,24 @@ public synchronized Part getParent() {
     public synchronized void setParent(Part parent) {
         this.parent = parent;
     }
+
+    /**
+     * Obtains the {@link StreamProvider} from the parent, if possible.
+     * Otherwise it obtains it from
+     * {@link Session#getDefaultInstance(java.util.Properties, Authenticator)}.
+     *
+     * @return the StreamProvider implementation from the session.
+     * @throws MessagingException if errors.
+     *
+     * @since JavaMail 2.2
+     */
+    protected StreamProvider getStreamProvider() throws MessagingException {
+        Part parent = this.parent;
+        if (parent != null) {
+            return parent.getStreamProvider();
+        } else {
+            return Session.getDefaultInstance(System.getProperties(), null).getStreamProvider();
```

Comment:
The else needs the nested catches:

````
try {
   try {
       return Session.getDefaultInstance(System.getProperties(), null).getStreamProvider();
    } catch (ServiceConfigurationError sce) {
        throw new IllegalStateException(sce);
    }
} catch (RuntimeException re) {
   throw new MessagingException("Unable to get " + StreamProvider.class.getName(), re);
}
`````

or needs to just invoke the super method.

Reply:
Multipart does not implement Part. We have to use your approach with the try/catch. It is in place.

Reply:
Now I need to re-review to understand that mult-part subparts correctly reach session of a message.